src/method/query/response.rs

```diff
@@ -0,0 +1,231 @@
+use std::ops::Deref;
+use std::ops::DerefMut;
+use std::ops::Index;
+use std::slice::SliceIndex;
+
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+
+use crate::Result;
+use crate::Value;
+
+/// A wrapper type around the list of results for the queries that were returned
+/// by the database.
+///
+/// Provides utility functions to access the result of one specific query, or if
+/// needed, all queries at once.
+///
+#[derive(Debug, Clone)]
+pub struct QueryResponse(Vec<QueryResult>);
+
+impl QueryResponse {
+    /// Constructs an empty [`QueryResponse`]
+    pub fn new() -> Self {
```

Comment:
Does this need to be public?

Reply:
I agree the function is not really needed. At the moment it is not used anywhere in `surrealdb.rs` except its [unit tests for the mocking](https://github.com/surrealdb/surrealdb.rs/pull/11/files#diff-a237413b1d328ab9f88ef959508c9126db825dfd4a0dbd898d2864e85d5ef569R71). And even there i suppose we could replace it with a `QueryResponse(Default::default())` or a `QueryResponse(Vec::new())` instead if we don't want a somewhat useless `new` function on the type.

Reply:
`pub(crate)` is fine.